Persephone/Controllers/QueueViewController.swift

```diff
@@ -0,0 +1,183 @@
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    queueView.columnAutoresizingStyle = .uniformColumnAutoresizingStyle
+
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(stateChanged(_:)),
+      name: MPDClient.stateChanged,
+      object: AppDelegate.mpdClient
+    )
+
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(queueChanged(_:)),
+      name: MPDClient.queueChanged,
+      object: AppDelegate.mpdClient
+    )
+
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(queuePosChanged(_:)),
+      name: MPDClient.queuePosChanged,
+      object: AppDelegate.mpdClient
+    )
+  }
+
+  @objc func stateChanged(_ notification: Notification) {
+    guard let state = notification.userInfo?[MPDClient.stateKey] as? mpd_state
+      else { return }
+
+    setQueueIcon(state)
+  }
+
+  @objc func queueChanged(_ notification: Notification) {
+    guard let queue = notification.userInfo?[MPDClient.queueKey] as? [MPDClient.Song]
+      else { return }
+
+    self.queue = queue
+
+    queueView.reloadData()
+  }
+
+  @objc func queuePosChanged(_ notification: Notification) {
+    guard let queuePos = notification.userInfo?[MPDClient.queuePosKey] as? Int32
+      else { return }
+
+    let oldSongRowPos = Int(self.queuePos + 1)
+    let newSongRowPos = Int(queuePos + 1)
+    self.queuePos = queuePos
+
+    setQueuePos(oldSongRowPos: oldSongRowPos, newSongRowPos: newSongRowPos)
+
+    queueView.reloadData(
+      forRowIndexes: [oldSongRowPos, newSongRowPos],
+      columnIndexes: [0, 1]
+    )
+  }
+
+  func setQueueIcon(_ state: mpd_state) {
+    switch state {
+    case MPD_STATE_PLAY:
+      self.queueIcon = NSImage(named: NSImage.Name("playButton"))
+    case MPD_STATE_PAUSE:
+      self.queueIcon = NSImage(named: NSImage.Name("pauseButton"))
+    default:
+      self.queueIcon = nil
+    }
+  }
+
+  func setQueuePos(oldSongRowPos: Int, newSongRowPos: Int) {
+    if oldSongRowPos > 0 {
+      guard let oldSongRow = queueView.rowView(atRow: oldSongRowPos, makeIfNecessary: true)
+        else { return }
+      guard let oldSongTitleCell = oldSongRow.view(atColumn: 0) as? NSTableCellView
+        else { return }
+
+      setRowFont(rowView: oldSongRow, font: systemFontRegular)
+      oldSongTitleCell.imageView?.image = nil
+    }
+
+    guard let songRow = queueView.rowView(atRow: newSongRowPos, makeIfNecessary: true)
+      else { return }
+    guard let newSongTitleCell = songRow.view(atColumn: 0) as? NSTableCellView
+      else { return }
+
+    setRowFont(rowView: songRow, font: systemFontBold)
+    newSongTitleCell.imageView?.image = self.queueIcon
+  }
+
+  func setRowFont(rowView: NSTableRowView, font: NSFont) {
+    guard let songTitleCell = rowView.view(atColumn: 0) as? NSTableCellView
+      else { return }
+    guard let songArtistCell = rowView.view(atColumn: 1) as? NSTableCellView
+      else { return }
+
```

Comment:
If you like, these could be combined into a single `guard let` statement with each `let` phrase separated by a comma.